src/commands/platforms/create.ts

```diff
@@ -99,37 +129,34 @@ export default class PlatformCreate extends BaseCommand {
 
     const account = await AccountUtils.getAccount(this.app, flags.account, { account_message: 'Select an account to register the platform with' });
 
-    const platform = await this.createArchitectPlatform(flags);
+    const contexts = await this.setupKubeContext(flags);
 
-    const platform_dto = { name: platform_name, ...platform, flags: flags_map };
+    try {
+      const platform = await this.createArchitectPlatform(flags, contexts.current_context);
 
-    CliUx.ux.action.start('Registering platform with Architect');
-    const created_platform = await this.postPlatformToApi(platform_dto, account.id);
-    CliUx.ux.action.stop();
-    this.log(`Platform registered: ${this.app.config.app_host}/${account.name}/platforms/new?platform_id=${created_platform.id}`);
+      const platform_dto = { name: platform_name, ...platform, flags: flags_map };
 
-    if (!flags['auto-approve']) {
-      const confirmation = await inquirer.prompt({
-        type: 'confirm',
-        name: 'application_install',
-        message: `Would you like to install the requisite networking applications? This is a required step before using Architect with this platform. More details at the above URL.`,
-      });
-      if (!confirmation.application_install) {
-        this.warn(`Installation cancelled. You will be unable to deploy services to this platform.\n\nIf you decide to proceed with installation, you can do so at the above URL. Or if you would like to deregister this platform from Architect, run: \n\narchitect platform:destroy -a ${account.name} --auto_approve ${platform_name}`);
-        return;
+      CliUx.ux.action.start('Registering platform with Architect');
+      const created_platform = await this.postPlatformToApi(platform_dto, account.id);
+      CliUx.ux.action.stop();
+      this.log(`Platform registered: ${this.app.config.app_host}/${account.name}/platforms/new?platform_id=${created_platform.id}`);
+
+      if (flags.type?.toLowerCase() == 'agent') {
+        CliUx.ux.action.start(chalk.blue('Installing the agent'));
+        await AgentPlatformUtils.installAgent(flags, created_platform.token.access_token, AgentPlatformUtils.getServerAgentHost(this.app.config.agent_server_host), this.app.config);
+        await AgentPlatformUtils.waitForAgent(flags);
+        CliUx.ux.action.stop();
+        await this.installAppliations(flags, created_platform, account.name, platform_name);
+      } else {
+        await this.installAppliations(flags, created_platform, account.name, platform_name);
```

Comment:
Seeing as this line is duplicated, I don't see why the `else` can't be removed. Then this line can just live under the `if` portion